SourceryFramework/Sources/Parsing/Composer.swift

```diff
@@ -348,6 +375,16 @@ public struct Composer {
             return nil
         }
     }
+    
+    private func specializeType(_ type: Type?, with genericParameters: [GenericTypeParameter]) -> Type? {
+        guard let type = type else { return nil }
+        // This hack with keyed archivers serves only one goal - copy type to specialize it with generic parameters
+        // We don't want to change original type, it should stay unspecialized
+        let data = NSKeyedArchiver.archivedData(withRootObject: type)
+        let typeCopy = NSKeyedUnarchiver.unarchiveObject(with: data) as? Type
```

Comment:
I guess this is done for brevity and not checking for all the possible types of `Type`, but this will probably come with a performance hit? Maybe would be better to introduce our custom Copyable protocol and generate code for all subclasses of `Type` that would use plain `init`.

Reply:
I agree, but i think this would be a bigger change, that needs to be resolved in a separate pull request. I'm adding a TODO item to those lines to address your comment.

Reply:
The problem is that I guess this may affect performance of processing source code for everyone even if generic types are not used after all. I can't tell for sure though, but we could check on the performance tests that that we should have? If there is a negative effect but we want to fix it separately maybe it will make sense to introduce "experimental" feature flag and do not parse generics unless it is enabled? Also considering that (I guess) generics support is still not complete for some things.

Reply:
I tried to run perfomance tests in `Sourcery+PerformanceSpec`, but they seem to be broken in a lot of ways, not sure how to properly fix and run them.

I tried to see perfomance difference on one of my fairly large projects, but got hit with some issues with indirect enums, that crash Sourcery on my project when used. Unfortunately, I rebased on master, so all features currently in master are in Sourcery build i tried.

Experimental flag seems a good idea to me, since there are some other things that could be done with generics, however I have no idea how to properly implement a feature flag for Sourcery. 

if you can try to run perfomance tests on Sourcery and/or on one of bigger project you have that use Sourcery, that would be great.